var/spack/repos/builtin/packages/cmake/package.py

```diff
@@ -132,6 +132,11 @@ class Cmake(Package):
     # https://gitlab.kitware.com/cmake/cmake/issues/18232
     patch('nag-response-files.patch', when='@3.7:3.12')
 
+    # Cray libhugetlbfs and icpc warnings failing CXX tests
+    # https://gitlab.kitware.com/cmake/cmake/-/merge_requests/4698
+    # https://gitlab.kitware.com/cmake/cmake/-/merge_requests/4681
+    patch('ignore_crayxc_warnings.patch', when='@3.7:3.17.2')
```

Comment:
@jennfshr Does this actually cleanly apply all the way back to 3.7?

Reply:
I looped over every CMake version from `spack info cmake` and attempted a build with the patch (since renamed) to see where it breaks, since I assumed it would.  I did carefully place the changes to the cmake/Source/Checks/cm_cxx_features.cmake file and test that the patches worked.  There's no v3.7.0 available via Spack, but I did test v3.7.1. 

I think that this is where the builds started to break with the libhugetlbfs and icpc warnings, and likely there are some circumstances where the patch is not doing anything in the context of the cm_cxx_features.cmake file:

```
0068224f introduced a feature that started causing issues:
C++ feature checks: check output for '[Ww]arning' June 3, 2017
eb3abf8b2ba514cec7d882bcde9d136fa3e023b7 C++ feature checks: Prepare for multiple check output filters
C++ feature checks: Prepare for multiple check output filters  Sept 17, 2017
```

I'm totally open to further paring down the list, if you know exactly what release tag this change corresponds to.

Here are the results of the tests I conducted on Trinitite. 

```
jgreen@tt-fey1:/usr/projects/hpctools/jgreen/pav2-lanl-new> grep -c "cmake: Successfully installed cmake"  /usr/projects/hpctools/jgreen/jgreen-cmake/cmake.spack*  |awk -F: '{if($2==0){print $1" FAILED"} }'|sed 's/\/usr\/projects\/hpctools\/jgreen\/jgreen-cmake\/cmake\.spack\.//'g
2.8.10.2-cce-9.1.1 FAILED
2.8.10.2-gcc-8.3.0 FAILED
2.8.10.2-intel-19.0.4 FAILED
3.0.2-cce-9.1.1 FAILED
3.0.2-gcc-8.3.0 FAILED
3.0.2-intel-19.0.4 FAILED
3.1.0-cce-9.1.1 FAILED
3.1.0-gcc-8.3.0 FAILED
3.1.0-intel-19.0.4 FAILED
3.11.0-intel-19.0.4 FAILED
3.11.1-intel-19.0.4 FAILED
3.11.2-intel-19.0.4 FAILED
3.11.3-intel-19.0.4 FAILED
3.11.4-intel-19.0.4 FAILED
3.17.0-intel-19.0.4 FAILED
3.3.1-cce-9.1.1 FAILED
3.3.1-gcc-8.3.0 FAILED
3.3.1-intel-19.0.4 FAILED
3.4.0-cce-9.1.1 FAILED
3.4.0-gcc-8.3.0 FAILED
3.4.0-intel-19.0.4 FAILED
3.4.3-cce-9.1.1 FAILED
3.4.3-gcc-8.3.0 FAILED
3.4.3-intel-19.0.4 FAILED
3.5.0-cce-9.1.1 FAILED
3.5.0-gcc-8.3.0 FAILED
3.5.0-intel-19.0.4 FAILED
3.5.1-cce-9.1.1 FAILED
3.5.1-gcc-8.3.0 FAILED
3.5.1-intel-19.0.4 FAILED
3.5.2-cce-9.1.1 FAILED
3.5.2-gcc-8.3.0 FAILED
3.5.2-intel-19.0.4 FAILED
3.6.0-cce-9.1.1 FAILED
3.6.0-gcc-8.3.0 FAILED
3.6.0-intel-19.0.4 FAILED
3.6.1-cce-9.1.1 FAILED
3.6.1-gcc-8.3.0 FAILED
3.6.1-intel-19.0.4 FAILED
```
```
jgreen@tt-fey1:/usr/projects/hpctools/jgreen/pav2-lanl-new> grep -c "cmake: Successfully installed cmake"  /usr/projects/hpctools/jgreen/jgreen-cmake/cmake.spack*  |awk -F: '{if($2==1){print $1" PASSED"} }'|sed 's/\/usr\/projects\/hpctools\/jgreen\/jgreen-cmake\/cmake\.spack\.//'g
3.10.0-cce-9.1.1 PASSED
3.10.0-gcc-8.3.0 PASSED
3.10.0-intel-19.0.4 PASSED
3.10.1-cce-9.1.1 PASSED
3.10.1-gcc-8.3.0 PASSED
3.10.1-intel-19.0.4 PASSED
3.10.2-cce-9.1.1 PASSED
3.10.2-gcc-8.3.0 PASSED
3.10.2-intel-19.0.4 PASSED
3.10.3-cce-9.1.1 PASSED
3.10.3-gcc-8.3.0 PASSED
3.10.3-intel-19.0.4 PASSED
3.11.0-cce-9.1.1 PASSED
3.11.0-gcc-8.3.0 PASSED
3.11.1-cce-9.1.1 PASSED
3.11.1-gcc-8.3.0 PASSED
3.11.2-cce-9.1.1 PASSED
3.11.2-gcc-8.3.0 PASSED
3.11.3-cce-9.1.1 PASSED
3.11.3-gcc-8.3.0 PASSED
3.11.4-cce-9.1.1 PASSED
3.11.4-gcc-8.3.0 PASSED
3.12.0-cce-9.1.1 PASSED
3.12.0-gcc-8.3.0 PASSED
3.12.0-intel-19.0.4 PASSED
3.12.1-cce-9.1.1 PASSED
3.12.1-gcc-8.3.0 PASSED
3.12.1-intel-19.0.4 PASSED
3.12.2-cce-9.1.1 PASSED
3.12.2-gcc-8.3.0 PASSED
3.12.2-intel-19.0.4 PASSED
3.12.3-cce-9.1.1 PASSED
3.12.3-gcc-8.3.0 PASSED
3.12.3-intel-19.0.4 PASSED
3.12.4-cce-9.1.1 PASSED
3.12.4-gcc-8.3.0 PASSED
3.12.4-intel-19.0.4 PASSED
3.13.0-cce-9.1.1 PASSED
3.13.0-gcc-8.3.0 PASSED
3.13.0-intel-19.0.4 PASSED
3.13.1-cce-9.1.1 PASSED
3.13.1-gcc-8.3.0 PASSED
3.13.1-intel-19.0.4 PASSED
3.13.2-cce-9.1.1 PASSED
3.13.2-gcc-8.3.0 PASSED
3.13.2-intel-19.0.4 PASSED
3.13.3-cce-9.1.1 PASSED
3.13.3-gcc-8.3.0 PASSED
3.13.3-intel-19.0.4 PASSED
3.13.4-cce-9.1.1 PASSED
3.13.4-gcc-8.3.0 PASSED
3.13.4-intel-19.0.4 PASSED
3.14.0-cce-9.1.1 PASSED
3.14.0-gcc-8.3.0 PASSED
3.14.0-intel-19.0.4 PASSED
3.14.1-cce-9.1.1 PASSED
3.14.1-gcc-8.3.0 PASSED
3.14.1-intel-19.0.4 PASSED
3.14.2-cce-9.1.1 PASSED
3.14.2-gcc-8.3.0 PASSED
3.14.2-intel-19.0.4 PASSED
3.14.3-cce-9.1.1 PASSED
3.14.3-gcc-8.3.0 PASSED
3.14.3-intel-19.0.4 PASSED
3.14.4-cce-9.1.1 PASSED
3.14.4-gcc-8.3.0 PASSED
3.14.4-intel-19.0.4 PASSED
3.14.5-cce-9.1.1 PASSED
3.14.5-gcc-8.3.0 PASSED
3.14.5-intel-19.0.4 PASSED
3.15.0-cce-9.1.1 PASSED
3.15.0-gcc-8.3.0 PASSED
3.15.0-intel-19.0.4 PASSED
3.15.1-cce-9.1.1 PASSED
3.15.1-gcc-8.3.0 PASSED
3.15.1-intel-19.0.4 PASSED
3.15.2-cce-9.1.1 PASSED
3.15.2-gcc-8.3.0 PASSED
3.15.2-intel-19.0.4 PASSED
3.15.3-cce-9.1.1 PASSED
3.15.3-gcc-8.3.0 PASSED
3.15.3-intel-19.0.4 PASSED
3.15.4-cce-9.1.1 PASSED
3.15.4-gcc-8.3.0 PASSED
3.15.4-intel-19.0.4 PASSED
3.15.5-cce-9.1.1 PASSED
3.15.5-gcc-8.3.0 PASSED
3.15.5-intel-19.0.4 PASSED
3.16.0-cce-9.1.1 PASSED
3.16.0-gcc-8.3.0 PASSED
3.16.0-intel-19.0.4 PASSED
3.16.1-cce-9.1.1 PASSED
3.16.1-gcc-8.3.0 PASSED
3.16.1-intel-19.0.4 PASSED
3.16.2-cce-9.1.1 PASSED
3.16.2-gcc-8.3.0 PASSED
3.16.2-intel-19.0.4 PASSED
3.16.5-cce-9.1.1 PASSED
3.16.5-gcc-8.3.0 PASSED
3.16.5-intel-19.0.4 PASSED
3.17.0-cce-9.1.1 PASSED
3.17.0-gcc-8.3.0 PASSED
3.7.1-cce-9.1.1 PASSED
3.7.1-gcc-8.3.0 PASSED
3.7.1-intel-19.0.4 PASSED
3.7.2-cce-9.1.1 PASSED
3.7.2-gcc-8.3.0 PASSED
3.7.2-intel-19.0.4 PASSED
3.8.0-cce-9.1.1 PASSED
3.8.0-gcc-8.3.0 PASSED
3.8.0-intel-19.0.4 PASSED
3.8.1-cce-9.1.1 PASSED
3.8.1-gcc-8.3.0 PASSED
3.8.1-intel-19.0.4 PASSED
3.8.2-cce-9.1.1 PASSED
3.8.2-gcc-8.3.0 PASSED
3.8.2-intel-19.0.4 PASSED
3.9.0-cce-9.1.1 PASSED
3.9.0-gcc-8.3.0 PASSED
3.9.0-intel-19.0.4 PASSED
3.9.2-cce-9.1.1 PASSED
3.9.2-gcc-8.3.0 PASSED
3.9.2-intel-19.0.4 PASSED
3.9.4-cce-9.1.1 PASSED
3.9.4-gcc-8.3.0 PASSED
3.9.4-intel-19.0.4 PASSED
3.9.6-cce-9.1.1 PASSED
3.9.6-gcc-8.3.0 PASSED
3.9.6-intel-19.0.4 PASSED
```

Reply:
Well that's certainly far better than many of the patches we see see, so yeah, I'm good with it.